### ISSUE TITLE
Externalize Annotations into slick jar

### ISSUE THREAD

**Issue by Xnyle** [quarkusio/quarkus#22589]
### Description

With Quarkus 2+ I need to put a io.quarkus.runtime.annotations.StaticInitSafe annotation on my ConfigSource if I need static init.

This annotation is part of core though and the library that contains the ConfigSource is not only used with quarkus.

YES i could add a compileOnly dependence to quarkus-code but that doesn't look right.

Quarkus should have a slick stereotype only dependency for the kind of stuff that might be needed to enrich other libs in order to make them function properly with quarkus.

**Comment by Sanne** [comment]
Very interesting use case. WDYT of this @radcortez ? Are we ready to support these as stable APIs ?

**Comment by radcortez** [comment]
Maybe...

In this particular case, `@StaticInitSafe` was added to allow registration of some config components in the static init config since we cannot trust that they will work. What I'm realizing, is that users require even more ways to customize their config instance, and adding annotations for all these cases is overkill.

Right now, internally in Quarkus I added build steps that allow you access to the `Config` builder in both static init and runtime, so users will be able to configure it exactly as they want. I see this as a better alternative to the use of annotations (which is limited and we lack support for).

Anyway, we need to get these APIs right if we want to expose them in such a way.
